packages/application-shell/src/components/custom-view-shell-authenticated/custom-view-shell-authenticated.tsx

```diff
@@ -63,14 +88,19 @@ function CustomViewShellAuthenticated(
                             if (isProjectLoading) {
                               return <ApplicationLoader />;
                             }
+
                             return (
                               <ApplicationContextProvider
                                 user={user}
                                 project={project}
                                 projectDataLocale={props.dataLocale}
                                 environment={props.environment}
                               >
-                                {props.children}
+                                <CustomViewWithPermissionCheck
```

Comment:
We make sure the user has, at least, view permission for the current custom view.